src/main/java/com/deque/axe/android/colorcontrast/ColorContrastRunner.java

```diff
@@ -140,6 +150,96 @@ ColorContrastResult onRowEnd() {
     return result;
   }
 
+  ColorContrastResult onRowEnd(AxeColor actualTextColor) {
+
+    ColorContrastResult result = new ColorContrastResult();
+
+    CountMap<ColorPair> pairsWithSimilarTextColor = new CountMap<>();
+
+    for (Map.Entry<ColorPair, Integer> entry1 : countExactPairs.entrySet()) {
+      for (Map.Entry<ColorPair, Integer> entry2 : countExactPairs.entrySet()) {
+        if (entry1.getKey().backgroundColor.equals(entry2.getKey().backgroundColor)) {
+          if (entry1.getKey().textColor.isVisiblySameColor(entry2.getKey().textColor)) {
+            pairsWithSimilarTextColor.increment(entry1.getKey(), entry2.getValue());
+          }
+        }
+      }
+    }
+
+    List<Map.Entry<ColorPair, Integer>> sortedByValueAndContrast = pairsWithSimilarTextColor
+            .entriesSortedByValue((o1, o2) -> o1.compareTo(o2));
+
+    List<Map.Entry<ColorPair, Integer>> filteredList = new ArrayList<>();
+
+    for (Map.Entry<ColorPair, Integer> entry : sortedByValueAndContrast) {
+      ColorPair colorPair = entry.getKey();
+      if (colorPair.backgroundColor.equals(actualTextColor)
+              || colorPair.textColor.equals(actualTextColor)) {
+        if (colorPair.backgroundColor.equals(actualTextColor)) {
+          filteredList.add(new Map.Entry<ColorPair, Integer>() {
+            @Override
+            public ColorPair getKey() {
+              return new ColorPair(entry.getKey().textColor, entry.getKey().backgroundColor);
+            }
+
+            @Override
+            public Integer getValue() {
+              return entry.getValue();
+            }
+
+            @Override
+            public Integer setValue(Integer value) {
+              return null;
+            }
+          });
+        } else {
+          filteredList.add(entry);
+        }
+      }
+    }
+
+    if (filteredList.size() <= 0) {
+      return result;
+    }
+
+    final Set<ColorPair> resultPairs = new HashSet<>();
+
+    Map.Entry<ColorPair, Integer> firstEntry = null;
+
+    for (Map.Entry<ColorPair, Integer> entry : filteredList) {
+
+      final ColorPair colorPair = entry.getKey();
+
+      final Integer occurrenceCount = entry.getValue();
+
+      if (occurrenceCount < ColorContrastConfig.MIN_CHARACTERS) {
+        break;
+      }
+
+      final Integer adjustedOccurrenceCount = occurrenceCount
+              * ColorContrastConfig.TRANSITION_COUNT_DOMINANCE_FACTOR;
+
+
+      if (firstEntry != null) {
+        if (adjustedOccurrenceCount < firstEntry.getValue()) {
+          break;
+        }
+      }
+
+      resultPairs.add(colorPair);
+
+      if (firstEntry == null) {
+        firstEntry = entry;
+      }
```

Comment:
In case of the first ever entry, the below logic will be executed on itself:
`if (firstEntry != null) {
        if (adjustedOccurrenceCount < firstEntry.getValue()) {
          break;
        }
      }`

Reply:
Is that actually a problem?  I don't think that `if (adjustedOccurenceCount < firstEntry.getValue())` will ever be `true` for the first entry unless `TRANSITION_COUNT_DOMINANCE_FACTOR` is less than 1?

`TRANSITION_COUNT_DOMINANCE_FACTOR = 2` (from ColorContrastConfig file)

`adjustedOccurrenceCount = entry.getValue() * 2` .... which means that the `if` statement will be checking `entry.getValue() * 2 < entry.getValue()`, which I don't think will ever be `true` for the first entry